[Xu Yi] iP

optionalemon opened the pull request:
# Pikachu Task Manager ϞϞ(๑⚈ ․̫ ⚈๑)∩ 🟡

> "Pokémon Can Evolve Into Better Versions Of Themselves. And Now - So Can You." 
> - Detective Pikachu 🟡v🟡

With Pikachu Task Manager, you can become a better version of yourself through managing your tasks better and never going to miss any deadlines! (**PIKA!**)

## Type of tasks it can manage:
* To do
* Event
* Deadline

All you need to do is,
1. download pikachu.jar from [here](https://github.com/optionalemon/ip/releases/tag/A-Jar)
2. double-click it
3. Add in your tasks
4. Pikachu will be there to help (*Pikapi!*)

If you are a fan of Pokémon:
- [ ] Yes, I love it!
- [ ] ~~No, but~~ Pikachu is cute!

And also a Java Programmer, you can definitely make use of the <code>main</code> function in Pikachu.java:
```java
public static void main(String[] args) {
        new Pikachu("you/path/to/database/file").run();
    }
```

Commit:
Add Gradle support

xyinxy committed:
level 1

xyinxy committed:
level 2

xyinxy committed:
level 3

xyinxy committed:
level 4

xyinxy committed:
automated text ui testing

xyinxy committed:
level 5

xyinxy committed:
level 6

xyinxy committed:
level 7

xyinxy committed:
level 8

xyinxy committed:
merge and solve conflic

xyinxy committed:
basic structure with moreOOP

sjoann commented on src/main/java/Deadline.java:
```diff
@@ -7,6 +7,14 @@ public Deadline(String description, String by) {
+        return "D";
+    }
+
+    public String timing() {
```
Perhaps it would be better to name the method such as getTiming() ?

optionalemon replied:
Thank you! Changed :)

sjoann commented on src/main/java/Duke.java:
```diff
@@ -56,8 +62,18 @@ static String validUnmark(String input) {
+            e.printStackTrace();
+        }
+        
         String intro = withFormat("Pika Pikachu! (I am Pikachu!)\nPi-ka-chu?(Do you need any help?)");
```
I like that you changed Duke's personality

sjoann reviewed (commented):
Overall great work ! Do remember to implement other stuff such as JavaDocs for the methods

xinbei-shen commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Perhaps this access modifier could be private since I do not see it being accessed outside of this class? Or is there any reason why you chose to use "protected"?

optionalemon replied:
I chose protected because I think both protected and private may work, and protected may help if any access is needed in the future code implementation HAHA. Thank you for your suggestion!

xinbei-shen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,169 @@
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch(NumberFormatException e){
```
perhaps include a spacing before and after () per the coding standard? :)

optionalemon replied:
Fixed, thank you!

xinbei-shen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,169 @@
+
+    static String validDelete(String input) {
+        if (!isNumeric(input.substring(7))) {
+            return "Pi-must be numbers behind-pi!";
```
i like the personality of your chatbot i think it is very cute!

xinbei-shen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,169 @@
+    }
+
+    static String validMark(String input) {
+        if (!isNumeric(input.substring(5))) {
+            return "Pi-must be numbers behind-pi!";
+        } else if (Integer.parseInt(input.substring(5)) > tasks.size() || Integer.parseInt(input.substring(5)) <= 0) {
+            return "Pi-not within range-pi!";
```
maybe these checks for valid input could be put into a separate method? since I noticed that the same checks are used in multiple other methods such as validDelete and validUnmark.

optionalemon replied:
That's true :( Will try to improve that in the future!

xinbei-shen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,169 @@
+            if (input.equals("bye")) { //bye
+                done = true;
+                tempStr = byebye;
+            } else if (input.equals("list")) { //list all items
```
I like that you put comments to make the code easier to understand

xinbei-shen reviewed (commented):
Great work! nice and readable code, and super cute chatbot personality 😃

xyinxy committed:
OOP class modified

xyinxy committed:
apply package format

xyinxy committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

* 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip:
  Add Gradle support

xyinxy committed:
set up gradle

xyinxy committed:
add junit tests

xyinxy committed:
fix datetime format bug

xyinxy committed:
change relative address to absolute address

jicsontoh commented on src/main/java/pikachu/Pikachu.java:
```diff
@@ -0,0 +1,53 @@
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch(NumberFormatException e){
```
Perhaps should have some spacing between the catch and the bracket as well as the curly brackets

optionalemon replied:
fixed, thank you!

jicsontoh commented on src/main/java/pikachu/command/AddCommand.java:
```diff
@@ -0,0 +1,60 @@
+    public AddCommand(String fullCommand) {
+        input = fullCommand;
+    }
+
```
Maybe can have some javadocs to explain the methods, especially for longer methods

optionalemon replied:
fixed, thank you!

jicsontoh commented on src/test/java/pikachu/task/EventTest.java:
```diff
@@ -0,0 +1,17 @@
+
+public class EventTest {
+    @Test
+    public void getNameTest(){
```
Java code normally use camelCase for method names e.g., testStringConversion but when writing test methods, sometimes another convention is used: whatIsBeingTested_descriptionOfTestInputs_expectedOutcome e.g., intDivision_zeroDivisor_exceptionThrown

optionalemon replied:
fixed after looking through the java coding standard, thank you for pointing out! I did not know about this :(

jicsontoh commented on src/main/java/pikachu/command/AddCommand.java:
```diff
@@ -0,0 +1,60 @@
+                Deadline newDDL = new Deadline(temp2[0],date);
+                tasks.taskList.add(newDDL);
+                tempStr = "Pikapi(added): " + newDDL + '\n';
+                tempStr += "Pikaaaaa: " + tasks.taskList.size() + (tasks.taskList.size() > 1 ? " tasks" : " task");
```
Perhaps for long line of codes, can break into 2 sentence!

optionalemon replied:
fixed! Thank you!

jicsontoh reviewed (commented):
Good job on keeping on task with the ip! All the best! :)

xyinxy committed:
add JavaDoc comments

xyinxy committed:
modify according to coding standard

xyinxy committed:
add find command

xyinxy committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

* branch-A-CodingStandard:
  modify according to coding standard

xyinxy committed:
fix index issue with find command

xyinxy committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

* branch-A-JavaDoc:
  add JavaDoc comments

# Conflicts:
#	src/main/java/pikachu/Ui.java
#	src/main/java/pikachu/task/Task.java
#	src/test/java/pikachu/task/EventTest.java

xyinxy committed:
add full stop for javadocs

xyinxy committed:
check style for all java files

xyinxy committed:
use gfmd in description

xyinxy committed:
modify readme

xyinxy committed:
create gui based on javafx tutorial

xyinxy committed:
implement GUI

xyinxy committed:
Change main class name

As the text-based design has its start at Pikachu class, and GUI starts in the Main class, it is crucial to change main class name to Main for the .jar file to run properly.

It makes the further implementation easier too.

optionalemon committed:
Merge pull request #2 from optionalemon/branch-Level-10

Change main class name

xyinxy committed:
Add assertion

The assert features have been used to ensure correctness of the input to the command and to check the status of the task list.

Adding assertion ensures that code functions as expected.

xyinxy committed:
Improve code quality

Moved all the boolean conditions to check error into variables with a clear name on the functions they serve and deleted all the unnecessary code.

These allow the function of the code to be more obvious to the readers and avoid confusing the readers on the purpose of the unused code.

optionalemon committed:
Merge pull request #3 from optionalemon/branch-A-Assertions

Add assertion

optionalemon committed:
Merge branch 'master' into branch-A-CodeQuality

optionalemon committed:
Merge pull request #4 from optionalemon/branch-A-CodeQuality

Improve code quality

xyinxy committed:
Detect duplicates

xyinxy committed:
Revert "Detect duplicates"

This reverts commit a99c79e66d5b4e6528c6e0f6aff666ee24a96272.

xyinxy committed:
Revert "Revert "Detect duplicates""

This reverts commit 2731fb0e4fc8692bdf4aa39fc1e207ad7f1caf68.

xyinxy committed:
Modify storage path to function on different computers

xyinxy committed:
add close function for GUI after saying bye

xyinxy committed:
Change pikachu image based on command

xyinxy committed:
improve GUI

xyinxy committed:
Upload Ui.png

xyinxy committed:
fix bugs and checkstyle

xyinxy committed:
Edit user guide

xyinxy committed:
fix minor bugs

xyinxy committed:
change gui to work on linux

xyinxy committed:
fix bugs and modify GUI

xyinxy committed:
Update README

xyinxy committed:
Update README.md
